### DISCUSSION TITLE
Auto-Discovery service in Cluster Internal Central Scans architecture

### DISCUSSION THREAD

**Discussion by mguletskii** [secureCodeBox/secureCodeBox#2437] [Q&A]
Hello secureCodeBox team.

First let me say "thank you" for all your work on this project.
 
We're integrating secureCodeBox solution to our infrastructure and prefer to use [Cluster Internal Central Scans architecture](https://www.securecodebox.io/docs/architecture/deployment_view#_cluster_internal_central_scans) (all ScansTypes and Scans stored in one specific namespace).

But Auto-Discovery service trying to create all scans in a service/pod namespace. It's required to have ScanTypes, Hooks and Hooks credentials in every namespace that we want to scan.

How we can use Auto-Discovery service in Cluster Internal Central Scans architecture? Or only way is create all additional entities in every scanning namespaces?

**Comment by Weltraumschaf** [comment]
Hi @mguletskii,

thanks for submitting this issue. Since, this is rather a question than a bug I convert it to a discussion, so that it'll be picked up easier by others later on.

**Comment by paraddise** [comment]
Hi, any updates? I think this will be cool feature and allow engineers to get `Health` of the cluster

**Comment by J12934** [comment]
Hi,
It's possible to implement but not straight forward.
The auto-discovery is some features which do not work properly across namespace boundaries, so creating scans in a central namespace is tricky (e.g. referencing image pull secrets to get images from a private registry and ownerReferences used to delete scans when the resource they are scanning got deleted)

If somebody wan't to pick this up and implement it we are happy to review and merge it but I don't think anybody from the core team has time to work on this right now.

**Reply by paraddise** [reply]
I think for the first time we can take `imagePullSecrets` from pod definition, as kubelete does it.
Now auto-discovery just unusable. I think the main point for auto-discovery is you configured it once in system namespace and after that you just annotate namespaces or resources. Otherwise you need to constantly copy every scan types, hooks, secrets, configs to each annotated namespace.
I will try to take a quick look to code and make kind of a epic and describe upcoming changes.
